rubygems/index.js

```diff
@@ -0,0 +1,230 @@
+       */
+      if (line.startsWith("gem")) {
+        // Split on # and remove the comments at the end of any lines
+        const lineWithoutComment = line.split('#')[0]
+        return acc.concat(lineWithoutComment)
+      }
+      return acc
+    }, [])
+    return extract
+  }
+
+  /**
+   * - input: a single entry from extractDependenciesFromManifest, input format is of 
+   * "gem <name of gem>, "version (optional)", specifiers (optional)
+   */
+  getSpec(pkg) {
+    if (typeof pkg === "object") return pkg
+
+    // Version seems to always come after the gem name, if it is specified. 
+    const pkgParts = pkg.split(",")
+    // remove gem, trim off white space, remove quotes, now we have the name
+    const name = pkgParts[0].replace('gem', '').trim().replace(/'/g, '').replace(/"/g, '')
+
+    // If there's nothing after the name, then return latest as version
+    if (!pkgParts[1]) {
+      return {
+        name,
+        operator: '==',
+        versionSpec: 'latest',
+        toString: () => `${name}@`
+      }
+    }
+
+    const re = /^('|")(==|=|>=|>|<=|~>)?\s*([0-9.]+)('|")$/
+    const match = pkgParts[1].trim().match(re)
+    if (!match) throw new Error(`unparseable pkg dependency ${pkgParts[1]}`)
+    const operator = match[2] || '='
+    const version = match[3]
+
+    return {
+      name,
+      operator,
+      versionSpec: version,
+      toString: () => `${name}@${operator}${version}`
+    }
+  }
+
+  // To list versions - hit https://rubygems.org/api/v1/versions/[gem name].json
+  // To get deps of specific version - hit https://rubygems.org/api/v2/rubygems/[gem name]/versions/[version].json
+  // to get latest version deps - hit https://rubygems.org/api/v2/rubygems/[gem name].json
+  async getDependencies(pkgSpec) {
+    let name
+    let version
+
+    try {
+      const resolved = await this.resolve(pkgSpec)
+      name = resolved.name
+      version = resolved.version
+    } catch (e) {
+      this.log.error(e)
+      // unable to resolve the given spec; no way to get the deps for this input
+      return []
+    }
+
+    let body
+    if (!version) {
+      ({ body } = await this.got(`https://rubygems.org/api/v2/rubygems/${name}.json`, {
+        responseType: "json"
+      }))
+    } else {
+      ({ body } = await this.got(
+        `https://rubygems.org/api/v2/rubygems/${name}/versions/${version}.json`,
+        {
+          responseType: "json"
+        }
+      ))
+    }
+
+    const dependencyKeys = Object.keys(body.dependencies)
+    const depRequirements = dependencyKeys.reduce((acc, key) => {
+      const runtime = body.dependencies[key]
+      return acc.concat(runtime.reduce((ac, dep) => {
+        // each dep is formatted like this
+        /**
+         * {
+              "name": "activerecord",
+              "requirements": "= 3.0.18"
+          }
+        */
+        const versionSplit = dep.requirements.split(' ')
```

Comment:
nit: since you are using the results of this split immediately, it would be helpful to name them here, e.g. `const [operator, version] = dep.requirements.split(' ')`. then the spec on L133-138 doesn't require backtracking